pynamodb/models.py

```diff
@@ -1211,21 +1260,32 @@ def _serialize(self, attr_map=False, null_check=True):
             if serialized is None:
                 continue
             if attr_map:
-                attrs[attributes][attr.attr_name] = {
-                    ATTR_TYPE_MAP[attr.attr_type]: serialized
-                }
+                attrs[attributes][attr.attr_name] = Model._serialize_value(attr, value)
             else:
                 if attr.is_hash_key:
                     attrs[HASH] = serialized
                 elif attr.is_range_key:
                     attrs[RANGE] = serialized
                 else:
-                    attrs[attributes][attr.attr_name] = {
-                        ATTR_TYPE_MAP[attr.attr_type]: serialized
-                    }
+                    attrs[attributes][attr.attr_name] = Model._serialize_value(attr, value)
         return attrs
 
     @classmethod
+    def _serialize_value(cls, attr, value):
+        """
+        Serializes a value for use with DynamoDB
+
+        :param value: a value to serialize
+        """
+        serialized = attr.serialize(value)
+        if serialized is None:
+            if not attr.null:
+                raise ValueError("Attribute '{0}': cannot be None".format(attr.attr_name))
+            return {NULL: True}
```

Comment:
can we just set `serialized = True` and let the `ATTR_TYPE_MAP` on line 1286 handle the null bit? seems nicer maybe? don't really know tbh